### DISCUSSION TITLE
Unit tests

### DISCUSSION THREAD

**Discussion by ptabandzelic** [databrickslabs/dbx#466] [Q&A]
Hey, is it recommended to use databricks-connect so i could run unit tests on cluster on databricks? If not, should i just create locally spark session, how about running unit tests inside CI? Should i run it on agents(github actions)?
What is recommended?

**Comment by renardeinside** [comment]
hi @ptabandzelic , 
please take a look at [this](https://dbx.readthedocs.io/en/latest/concepts/testing/) guidance first.
In short, unit tests are something you should be running locally, without using Databricks clusters. 
As for integration tests (that are expected to be a part of your CI pipeline), please read the guidance [here](https://dbx.readthedocs.io/en/latest/guides/python/integration_tests/).
